### ISSUE TITLE
Remove all public-facing icons

### ISSUE THREAD

**Issue by carstingaxion** [GatherPress/gatherpress#631]
### Describe your question

As much as I appreciate the subtle Icons next to the event date, the address or the online event link, but I have  to suggest, to remove all of them, unfortunately.

While working on all the new block-variations I had to remove them one by one, as I came through the blocks, because I would had have to create `render_block` and `blockEdit` filters for each icon as block variations don’t support those kind of extra-markup out of the box.

Additionally I think not providing any frontend icons would allow GatherPress to be compatible with more themes which could provide their own icon sets. 

Don’t get me wrong, I like icons. And I do like the used dashicons. So maybe we could provide some nice screenshots using the [Icon Block](https://wordpress.org/plugins/icon-block/) Plugin from Nick Diego. Accompanied by some explanation, that Icons might be theme-territory and a link to the Icon plugin.

Alternatively we could make use of emoji within our provided default texts or at least highlight this possibility and a code example within the README. I tested this within my [interactive Add-to-calendar](https://github.com/GatherPress/gatherpress/issues/606) exploration.

![Bildschirmfoto vom 2024-03-17 16-13-44](https://github.com/GatherPress/gatherpress/assets/198883/a21b66d4-6c9e-4103-b51d-55f93e4cdeac)

_**EDIT: 01.04.2024**_

I totally forgot to mention the almost 60kb GatherPress is loading on every frontend-request, this is huge and disproportionately high compared to the 3-4 icons a user might, maybe see. _You can see this expensive call in the screenshot, too._

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by nilovelez** [comment]
These are the icons that are used now:

- dashicons-nametag: used as GatherPress logo and menu icon
- dashicons-location: for venue address
- dashicons-phone: for venue phone number
- dashicons-admin-site-alt3: globe icon for venue website
- dashicons-clock: for event time and date
- dashicons-calendar: for the "add to calendar button"
- dashicons-video-alt2: for "online event" text.

This is what I propose:

- video, clock and calendar: these are de only ones that might be needed to provide visual clues and a consistent user experience. They can be embedded into a custom-base64encoded font (I can do it ). It will reduce the payload to something around 2Kb
- dashicons-nametag: this can be loaded as an independent .svg file. It is only used in the backend.
- location, phone and web. Those are  unnecessary. The associated text is quite obvious.

**Comment by carstingaxion** [comment]
> - video, clock and calendar: these are de only ones that might be needed to provide visual clues and a consistent user experience. They can be embedded into a custom-base64encoded font (I can do it ). It will reduce the payload to something around 2Kb

Sound like a very good idea.
We _just_ need to find a consistent, editable or, in other words, user-friendly way to add the icons to the editor and the frontend. ;-)

> - dashicons-nametag: this can be loaded as an independent .svg file. It is only used in the backend.

I think we don't need to do anything about the nametag icon. 
Because only the backend uses it, can't we just rely on the dashicons that are loaded by default by core?

> - location, phone and web. Those are  unnecessary. The associated text is quite obvious.

Absolutely. 100%. No additions.

**Comment by carstingaxion** [comment]
> * They can be embedded into a custom-base64encoded font (I can do it ).

@nilovelez I'm now at the point, where it would be nice to make use of the icon font you suggested. For the new Add-to-calendar block, I created a variant with an (ugly, old demo) icon using a block-style, which seems to be the easiest and most user-friendly way to add icons to blocks.

![grafik](https://github.com/GatherPress/gatherpress/assets/198883/58678108-9109-4ee7-9cb9-84c6b6533862)

_But no stress._ I'm out for family holiday until the 09. of july. But in 2 weeks, I'd like to finish the new Add-to-calendar block with a nice icon. Could you prepare the font, please @nilovelez ?

**Comment by nilovelez** [comment]
Ok, I'll prepare a base64-encoded font with the video, clock and calendar
icons

**Comment by mauteri** [comment]
Left this in Slack, but going to leave it here too.

Let's hold off for now on creating any new icons based on feedback from accessibility call. I think we may take the advice to remove the icons, but I want to bring it to discussion and not just have it as one person's opinion, even if the icons end up different and are no longer affecting accessibility. I still want designs to look good.

**Comment by MervinHernandez** [comment]
Agree with Mike. Let's keep this in backlog for now.

**Comment by carstingaxion** [comment]
> Let's hold off for now on creating any new icons based on feedback from accessibility call. I think we may take the advice to remove the icons, but I want to bring it to discussion and not just have it as one person's opinion, even if the icons end up different and are no longer affecting accessibility.

Until here I can follow and also agree: Let's remove all public icons.

> I still want designs to look good.

But here I become uncomfortable with my interpretation. Keep or remove, what did you actually mean @mauteri ?

(Btw. I'd love to read the accessibility feedback GatherPress got.)

**Comment by mauteri** [comment]
@carstingaxion Mervin is currently getting the feedback into issues for us to review. Mostly what I was saying on this is folks we spoke to about accessibility mentioned removing the icons on our blocks, there was discussion here about doing it differently, and I just want to punt the conversation to later and not execute on anything concerning icons with blocks until we discuss it a bit further.

**Comment by carstingaxion** [comment]
Ok, thank you for your fast response @mauteri !
